ArduPlane/events.cpp

```diff
@@ -316,6 +320,14 @@ void Plane::handle_battery_failsafe(const char *type_str, const int8_t action)
                     plane.mission.set_in_landing_sequence_flag(true);
                     break;
                 }
+#if MODE_AUTOLAND_ENABLED
+            if ((Failsafe_Action)action == Failsafe_Action_AUTOLAND_OR_RTL) {
+                if (!set_mode(mode_autoland, ModeReason::BATTERY_FAILSAFE)) {
+                    set_mode(mode_rtl, ModeReason::BATTERY_FAILSAFE);
+                }
+            break;
```

Comment:
It will fall through to RTL in anycase. So you can combine the ifs.
```suggestion
            if (((Failsafe_Action)action == Failsafe_Action_AUTOLAND_OR_RTL) && set_mode(mode_autoland, ModeReason::BATTERY_FAILSAFE)) {              
                break;
```

Reply:
changed